core-web/apps/dotcms-ui/src/app/portlets/dot-edit-page/content/components/dot-edit-page-state-controller/dot-edit-page-state-controller.component.ts

```diff
@@ -226,20 +234,24 @@ export class DotEditPageStateControllerComponent implements OnChanges {
     private showConfirmation(): Observable<DotConfirmationType> {
         return from(
             new Promise<DotConfirmationType>((resolve, reject) => {
-                if (this.shouldAskToLock()) {
-                    this.showLockConfirmDialog()
-                        .then(() => {
-                            resolve(DotConfirmationType.LOCK);
-                        })
-                        .catch(() => reject());
-                }
-
                 if (this.shouldAskPersonalization()) {
                     this.showPersonalizationConfirmDialog()
                         .then(() => {
                             resolve(DotConfirmationType.PERSONALIZATION);
                         })
                         .catch(() => reject());
+                } else if (this.shouldAskOnRunningExperiment()) {
+                    this.showRunningExperimentConfirmDialog()
+                        .then(() => {
+                            resolve(DotConfirmationType.RUNNING_EXPERIMENT);
+                        })
+                        .catch(() => reject());
+                } else if (this.shouldAskToLock()) {
+                    this.showLockConfirmDialog()
+                        .then(() => {
+                            resolve(DotConfirmationType.LOCK);
+                        })
+                        .catch(() => reject());
```

Comment:
Is there a scenario where none of these conditions are true? If that's the case, we should include an `else { reject() }` statement here otherwise this will never get resolved.

Reply:
yes it is, but that is checked before call this function. so all good.